### PR TITLE
Add request-id to search operator

### PR DESCRIPTION
**Pull request by gingerwizard**
This allows a user to specify a request-id to a search operator. This will be passed down to the runner from the parameter source and include in metrics if specified. @danielmitterdorfer if you're happy with this approach ill push docs.

### CODE REVIEW

File: esrally/driver/runner.py [hunk 1]
```diff
@@ -951,28 +959,22 @@ async def request_body_query(self, es, params):
...
-                "timed_out": timed_out,
-                "took": took
-            }
-        else:
-            return {
-                "weight": 1,
-                "unit": "ops",
-                "success": True
-            }
+                "request_id": request_id
```
Comment (danielmitterdorfer) [inline esrally/driver/runner.py:966]: Is this a leftover? Because in the next line, the `request_id` is conditionally added to this dict.

**Review by danielmitterdorfer — COMMENTED** [review]
Nice addition! I think there is one leftover but apart from that it looks fine. Can you please add the new parameter `request-id` also to the operation docs in the track reference? I'll take care in https://github.com/elastic/rally/pull/1169 that the additional meta-data property is documented.

File: esrally/driver/runner.py (hunk 1 above)
Comment (gingerwizard) [reply esrally/driver/runner.py:966]: Well spotted and also a lint issue

**Review by dliappis — APPROVED** [review]
LGTM

File: esrally/driver/runner.py [hunk 2]
```diff
@@ -775,7 +775,11 @@ class Query(Runner):
     * `pages`: Number of pages to retrieve at most for this scroll. If a scroll query does yield less results than the specified number of
                pages we will terminate earlier.
     * `results-per-page`: Number of results to retrieve per page.
+    * `request-id`: a user specified string identifying the query. This will be included in rally metrics to allow query
+                execution to be tracked.
+
```
Comment (dliappis) [inline esrally/driver/runner.py:780]: @gingerwizard as mentioned by the lint failure in CI, this newline needs to go away.

**Comment by gingerwizard** [comment]
Discussed this with @danielmitterdorfer this doesn't bring us much sadly. For normal searches, we already index the operation name in the `operation` field. The composite operation doesn't propagate meta data and provide metadata for the searches inside it. This is therefore not that useful. Closing.

**Pull request closed without merging**